src/main/webapp/SearchResults.js

```diff
@@ -29,162 +59,67 @@ function loading() {
-          ' 38m Christ the Redeemer statue atop Mount Corcovado and for' +
-          ' Sugarloaf Mountain, a granite peak with cable cars to its summit.',
-      id: 202,
-    },
-    {
-      country: 'Brazil',
-      name: 'Langkawi',
-      description: 'Salvador, the capital of Brazil’s' +
-          ' northeastern state of Bahia, is known for its Portuguese colonial' +
-          ' architecture, Afro-Brazilian culture and a tropical coastline.',
-      id: 203,
-    },
-  ];
-  displayResults(results);
-}
-
-/**
- * Loads South Korea info to page.
- */
-function southKoreaResults() {
-  const headingContainer = document.getElementById('Heading');
-  headingContainer.innerText = 'SOUTH KOREA';
-
-  const results = [
-    {
-      country: 'South Korea',
-      name: 'Seoul',
-      description: 'Seoul, the capital of South Korea,' +
-          ' is a huge metropolis where modern skyscrapers, high-tech subways' +
-          ' and pop culture meet Buddhist temples, palaces and street markets.',
-      id: 201,
-    },
-    {
-      country: 'South Korea',
-      name: 'Busan',
-      description: 'Busan, a large port city in South' +
-          ' Korea, is known for its beaches, mountains and temples.',
-      id: 202,
-    },
-    {
-      country: 'South Korea',
-      name: 'Jeju-do',
-      description: 'Jeju province encompasses the ' +
-          'South Korean island of Jeju in the Korea Strait. Its known for' +
-          'its beach resorts and volcanic landscape of craters and cavelike' +
-          ' lava tubes.',
-      id: 203,
-    },
-  ];
-  displayResults(results);
-}
-
 /**
  * @param {results} results
  */
-function displayResults(results) {
-  // Create a flex-container div if necessary or grab last flex-container
+function displayCityResults(results, countries) {
   const flexElement = document.getElementById('flex-container');
 
   for (let result = 0, l = results.length; result < l; ++result) {
-    // Create a container div
+    if (results[result] !== undefined) {
     const divElement = document.createElement('div');
 
-    // Place image in div container
     const imgUrl = 'images.jpg';
     const imgElement = document.createElement('img');
     imgElement.src = imgUrl;
 
     divElement.appendChild(imgElement);
 
-    // Create a container div
     const divInfoElement = document.createElement('div');
     divInfoElement.className = 'container';
 
-    // populate city name
     const cityElement = document.createElement('h1');
     cityElement.innerText = results[result]['name'];
 
-    // populate country name
+    var countryName ="";
+    for (let country in countries) {
+        if(countries[country]['id'] === results[result]['countryId']){
+          var countryName = countries[country]['name'];
+          break;
+        }
+    }
+
     const countryElement = document.createElement('h2');
-    countryElement.innerText = results[result]['country'];
+    countryElement.innerText = countryName;
```

Comment:
This is interesting! I hadn't thought about adding the name of the country a city's in like this but it makes a lot of sense. Hmm, assuming you use the Search API I mentioned I can't think of a good way to do this unless you do something like this:

1. In the `searching()` function change the last suggested line to:
```javascript
const searchResponse = await fetch('/api/search?name=' + encodeURI(input));
const results = await searchResponse.json();
const countriesResponse = await fetch('/api/countries');
const countries = await countriesResponse.json();
const citiesResponse = await fetch('/api/cities');
const cities = await countriesResponse.json();
displayResults(results, cities, countries);
```

1. Change the method signature of `displayResults` to take in those three parameters.

1. Change the code I highlighted here to something like:
```javascript
    if (result.type == 'City') {
      let city = cities.filter(c => c.id == result.id)[0];
      let country = countries.filter(c => c.id == city.countryId)[0];
      const countryElement = document.createElement('h2');
      countryElement.innerText = country.name;
      divInfoElement.appendChild(countryElement);
    }
```

You'll need to delete the `divInfoElement.appendChild(countryElement);` statement below.

Note that the two filtering lines are assuming it can find the city/country.